applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_linear_elastic_plane_strain_2D_law.cpp

```diff
@@ -0,0 +1,201 @@
+
+    Matrix constitutive_matrix;
+    parameters.SetConstitutiveMatrix(constitutive_matrix);
+
+    Properties properties;
+    properties.SetValue(YOUNG_MODULUS, 1.0e7);
+    properties.SetValue(POISSON_RATIO, 0.3);
+    parameters.SetMaterialProperties(properties);
+
+    rConstitutiveLaw.CalculateMaterialResponsePK2(parameters);
+
+    return stress;
+}
+
+GeoLinearElasticPlaneStrain2DLaw CreateLinearElasticPlaneStrainLaw()
+{
+    return GeoLinearElasticPlaneStrain2DLaw{std::make_unique<PlaneStrainType>()};
+}
+
+} // namespace
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawClone, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto       law   = CreateLinearElasticPlaneStrainLaw();
+    const auto clone = law.Clone();
+    KRATOS_EXPECT_NE(&law, clone.get());
+    KRATOS_EXPECT_NE(dynamic_cast<const GeoLinearElasticPlaneStrain2DLaw*>(clone.get()), nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawRequiresInitializeResponse, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+    KRATOS_EXPECT_TRUE(law.RequiresInitializeMaterialResponse())
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawRequiresFinalizeResponse, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+    KRATOS_EXPECT_TRUE(law.RequiresFinalizeMaterialResponse())
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawIsIncremental, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+    KRATOS_EXPECT_TRUE(law.IsIncremental())
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawReturnsExpectedLawFeatures, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+
+    ConstitutiveLaw::Features law_features;
+    law.GetLawFeatures(law_features);
+
+    KRATOS_EXPECT_TRUE(law_features.mOptions.IsDefined(ConstitutiveLaw::PLANE_STRAIN_LAW))
+    KRATOS_EXPECT_TRUE(law_features.mOptions.IsDefined(ConstitutiveLaw::INFINITESIMAL_STRAINS))
+    KRATOS_EXPECT_TRUE(law_features.mOptions.IsDefined(ConstitutiveLaw::ISOTROPIC))
+
+    const auto& strain_measures = law_features.mStrainMeasures;
+    KRATOS_EXPECT_TRUE(std::find(strain_measures.begin(), strain_measures.end(),
+                                 ConstitutiveLaw::StrainMeasure_Infinitesimal) != strain_measures.end())
+    KRATOS_EXPECT_TRUE(std::find(strain_measures.begin(), strain_measures.end(),
+                                 ConstitutiveLaw::StrainMeasure_Deformation_Gradient) !=
+                       strain_measures.end())
+
+    KRATOS_EXPECT_EQ(law_features.mStrainSize, 4);
+    KRATOS_EXPECT_EQ(law_features.mSpaceDimension, 2);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawReturnsExpectedStrainSize, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
```

Comment:
I believe `law` can be made `const`:
```suggestion
    const auto law = CreateLinearElasticPlaneStrainLaw();
```